fahari/mle/models.py

```diff
@@ -0,0 +1,85 @@
+from django.contrib.gis.db import models
+from django.urls.base import reverse
+from django.utils import timezone
+
+from fahari.common.models import AbstractBase, Facility
+
+
+class FacilityNetworkCondition(AbstractBase):
+    """Facility network conditions."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    TRUE = "TRUE"
+    FALSE = "FALSE"
+    STATE_INDICATORS = [
+        (TRUE, "True"),
+        (FALSE, "False"),
+    ]
+    has_network = models.CharField(max_length=50, choices=STATE_INDICATORS, default=TRUE)
+    has_internet = models.CharField(max_length=50, choices=STATE_INDICATORS, default=TRUE)
+
+    def __str__(self) -> str:
+        return "Facility: %s, Has network: %s, Has internet: %s" % (
+            self.facility.name,
+            self.has_network,
+            self.has_internet,
+        )
+
+    def get_absolute_url(self):
+        update_url = reverse("mle:network_condition_update", kwargs={"pk": self.pk})
+        return update_url
+
+    class Meta:
+        unique_together = ("facility",)
+        ordering = (
+            "-updated",
+            "facility__name",
+        )
+
+
+class FacilityDevice(AbstractBase):
+    """Facility devices ."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    number_of_devices = models.IntegerField(default=0)
```

Comment:
For ease of validation. Change this to a `models.PositiveSmallIntegerField`.

Reply:
fixed